app/controllers/feedback_messages_controller.rb

```diff
@@ -19,12 +18,24 @@ def create
       )
       rate_limiter.track_limit_by_action(:feedback_message_creation)
 
-      redirect_to feedback_messages_path
+      respond_to do |format|
+        format.html { redirect_to feedback_messages_path }
+        format.json { render json: { success: true, message: "Your report is submitted" } }
+      end
     else
       @previous_message = feedback_message_params[:message]
-
       flash[:notice] = "Make sure the forms are filled 🤖"
-      render "pages/report_abuse"
+
+      respond_to do |format|
+        format.html { render "pages/report_abuse" }
+        format.json do
+          render json: {
+            success: false,
+            message: @feedback_message.errors.full_messages.to_sentence,
```

Comment:
You can do `@feedback_message.errors_as_sentence` here, we defined that on `ApplicationRecord`.